devices/temp_nix.go

```diff
@@ -28,6 +32,35 @@ func getTemps(temps map[string]int) map[string]error {
 			temps[label] = int(sensor.Temperature)
 		}
 	}
+
+	block, err := ghw.Block()
+	if err != nil {
+		log.Print("error getting block device info")
+		return nil
+	}
+
+	for _, disk := range block.Disks {
+		dev, err := smart.Open("/dev/" + disk.Name)
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+		defer dev.Close()
+
+		switch sm := dev.(type) {
+		case *smart.SataDevice:
+			data, _ := sm.ReadSMARTData()
```

Comment:
do not forget to handle error here

Reply:
good catch, I forgot to add checks after the ReadSmart() calls. Fixed in [3c148a4](https://github.com/xxxserxxx/gotop/pull/221/commits/3c148a4d97d0f54f1d6f464ddc64b03fda1ad5fa)

Reply:
it still does not use `err` variable. The line above must be `data, err := sm.ReadSMARTData()`

Reply:
oops! thanks for pointing it out! fixed in b89c1dd748b8ee0913f99cdb16079e0515ae7aba